### PR TITLE
PXP-7894: change drs compact uri

### PR DESCRIPTION
**Pull request by MaribelleHGomez**
Jira Ticket: [PXP-7894](https://ctds-planx.atlassian.net/browse/PXP-7894)

Change format of uri for drs objects from hostname to compact based

**Hostname**: drs://drs.example.org/314159
**Compact**: drs://[provider_code/]namespace:accession
**Example Compact Identifier**: drs://dg.4dfc:81944ba1-81d0-436e-8552-33d77a27834b

namespace = prefix
accession = uuid of object
we don’t currently use provider_code

### New Features
- Change format of uri for drs objects from hostname to compact based

### CODE REVIEW

File: indexd/drs/blueprint.py [hunk 1]
```diff
@@ -101,7 +101,20 @@ def indexd_to_drs(record, expand=False):
         else record["bundle_id"]
     )
 
-    self_uri = "drs://" + flask.current_app.hostname + "/" + did
+    defaultPrefix = blueprint.index_driver.config.get("DEFAULT_PREFIX")
+    accession = did
+    if "/" in did:
```
Comment (MaribelleHGomez) [inline indexd/drs/blueprint.py:106]: Replace with regex?

File: indexd/drs/blueprint.py [hunk 2]
```diff
@@ -101,7 +101,20 @@ def indexd_to_drs(record, expand=False):
...
+        namespace = did.split("/")[0]
+        accession = did.split("/")[1]
+    elif ":" in did:
+        namespace = did.split(":")[0]
+        accession = did.split(":")[1]
+    else:
+        namespace = "NO_DEFAULT" if not defaultPrefix else defaultPrefix
+
+    print(namespace)
```
Comment (MaribelleHGomez) [inline indexd/drs/blueprint.py:115]: Remove this :|

**Comment by BinamB** [comment]
You can add this in the top of the PR description:
```
Hostname: drs://drs.example.org/314159
Compact: drs://[provider_code/]namespace:accession
Example Compact Identifier: drs://dg.4dfc:81944ba1-81d0-436e-8552-33d77a27834b

namespace = prefix
accession = uuid of object
we don’t currently use provider_code
```
And make a bullet point for `Change format of uri for drs objects from hostname to compact based`

File: indexd/drs/blueprint.py [hunk 3]
```diff
@@ -101,7 +101,20 @@ def indexd_to_drs(record, expand=False):
...
+    defaultPrefix = blueprint.index_driver.config.get("DEFAULT_PREFIX")
+    accession = did
+    if "/" in did:
+        namespace = did.split("/")[0]
+        accession = did.split("/")[1]
+    elif ":" in did:
+        namespace = did.split(":")[0]
+        accession = did.split(":")[1]
+    else:
+        namespace = "NO_DEFAULT" if not defaultPrefix else defaultPrefix
```
Comment (MaribelleHGomez) [inline indexd/drs/blueprint.py:113]: What do we want for env with no default prefix configured?

File: tests/test_drs.py [hunk 4]
```diff
@@ -56,7 +56,7 @@ def test_drs_get(client, user):
         assert rec_2["checksums"][0]["checksum"] == data["hashes"][k]
         assert rec_2["checksums"][0]["type"] == k
     assert rec_2["version"]
-    assert rec_2["self_uri"] == "drs://fictitious-commons.io/" + rec_1["did"]
+    assert rec_2["self_uri"] == "drs://testprefix:" + rec_1["did"].split(":")[1]
```
Comment (MaribelleHGomez) [inline tests/test_drs.py:59]: seems odd

**Comment by BinamB** [comment]
Just as a reminder and so that you dont have to go back in our chat. Here's like the general logic of what we could do to clean this up a little. 

```
guids = ["0000047f-772f-4241-980d-1f667686fe60", "dg.4DFC/0000047f-772f-4241-980d-1f667686fe61", "dg.4DFC:0000047f-772f-4241-980d-1f667686fe62"]
prefix = "dg.4DFC"
for g in guids:
    if prefix in g:
        a = g.replace(prefix, "", 1)[1:]
        print("drs://" + prefix + ":" + a)
```

File: indexd/drs/blueprint.py [hunk 5]
```diff
@@ -84,6 +84,35 @@ def get_signed_url(object_id, access_id):
     return res, 200
 
 
+def create_drs_uri(did):
```
Comment (MaribelleHGomez) [inline indexd/drs/blueprint.py:87]: I split it out into it's own function since it's used more than once, lmk any thoughts

Comment (BinamB) [reply indexd/drs/blueprint.py:87]: This is fine. Feels cleaner

**Review by BinamB — APPROVED** [review]
lgtm! :rocket:

**Pull request merged**